### PR TITLE
Deprecated retryWhen usage substituted

### PR DESCRIPTION
**Pull request by sergiusz-n**
A few deprecated (as of v.3.3.x) `.retryWhen()`usages are updated. . The latter prevents from updating dependancy reactor-core to  [v.3.4.0](https://github.com/reactor/reactor-core/releases/tag/v3.4.0).

### CODE REVIEW

**Comment by alzimmermsft** [comment]
@srnagar @kushagraThapar could you review this PR, this looks to resolve issue #17685

**Review by kushagraThapar — APPROVED** [review]
LGTM, however, it would be great if someone else can review the specifics of `Retry.withThrowable()` and `Retry.from()`?
@anuchandy @srnagar

**Comment by srnagar** [comment]
The changes look good to me. Thank you for your contribution @sergiusz-n. We are about to release the core-amqp library and we want to hold off on merging this change until the release is complete. This change can be included for the subsequent release.

@kushagraThapar can you please check why cosmos live tests are failing? Is this related to the changes in this PR?

**Comment by kushagraThapar** [comment]
@srnagar  @alzimmermsft  - I am seeing this error in the build step -> 
```
##[error]Unhandled: Cannot read property 'major' of null
```

In the past, I have ran several live tests on open source contributions, but seeing this issue for the first time. Can you guys please check what might be causing this ?
I am not re-running the build to avoid loss of logs.

**Comment by alzimmermsft** [comment]
> @srnagar @alzimmermsft - I am seeing this error in the build step ->
> 
> ```
> ##[error]Unhandled: Cannot read property 'major' of null
> ```
> 
> In the past, I have ran several live tests on open source contributions, but seeing this issue for the first time. Can you guys please check what might be causing this ?
> I am not re-running the build to avoid loss of logs.

Looked into it an found that a new testing variable was added which wasn't present in the Cosmos testing infrastructure. I've submitted this PR (https://github.com/Azure/azure-sdk-for-java/pull/17762) to resolve this issue by adding a safe global default in case there is other infrastructure that is also missing the value.

**Comment by aderri** [comment]
Hi Guys,

I'm little bit stuck with this issue. Do you have an idea when it's going to be released please ?

Thanks in advance,
Anass

**Comment by kushagraThapar** [comment]
> Hi Guys,
> 
> I'm little bit stuck with this issue. Do you have an idea when it's going to be released please ?
> 
> Thanks in advance,
> Anass

The release pipeline is failing because of intermittent connectivity issues, we will merge the PR once all the pipelines finish up.

**Review by alzimmermsft — APPROVED** [review]

**Comment by aderri** [comment]
Hi Guys,

As the pipeline seems to be ok . Could anyone merge this PR please to release the fix ?

Have a nice day,
Anass.

**Comment by aderri** [comment]
> > Hi Guys,
> > I'm little bit stuck with this issue. Do you have an idea when it's going to be released please ?
> > Thanks in advance,
> > Anass
> 
> The release pipeline is failing because of intermittent connectivity issues, we will merge the PR once all the pipelines finish up.

Hi, 

Could you please accept MR as everything is green now please ?

Thanks,
Anass.

**Comment by aderri** [comment]
Hello guys,

Are you sure to have released this fix ?  Because I'm using the latest version (7.0.0) of Azure SDK for Java that includes the  2.0.0 version of amqp. I tried to find this fix but failed.

Could you help me please ?

Anass.

**Pull request merged**
